src/ast/semantic_analyser.cpp

```diff
@@ -332,6 +333,26 @@ void SemanticAnalyser::visit(Call &call)
           call.loc);
   }
 
+  struct func_setter
```

Comment:
This is fine for now, but if we use this pattern later anywhere else it'd be nice to pull it into util.h and make a generic RAII helper.

Reply:
Agreed. A few more examples will make that easier.